.circleci/config.yml

```diff
@@ -67,16 +54,35 @@ workflows:
               ignore: /.*/
             tags:
               only: '/^v[\d.]+$/'
+  nightly:
+    triggers:
+      - schedule:
+          cron: 0 7 * * *
+          filters:
+            branches:
+              only: master
+    jobs: *workflow_jobs
 jobs:
   node4:
     docker:
       - image: 'node:4'
         user: node
-    steps:
+    steps: &unit_tests_steps
       - checkout
+      - run: &remove_package_lock
+          name: Remove package-lock.json if needed.
+          command: |
+            WORKFLOW_NAME=`python .circleci/get_workflow_name.py`
+            echo "Workflow name: $WORKFLOW_NAME"
+            if [ "$WORKFLOW_NAME" = "nightly" ]; then
+              echo "Nightly build detected, removing package-lock.json"
+              rm -f package-lock.json samples/package-lock.json
+            else
+              echo "Not a nightly build, skipping this step."
+            fi
       - run:
           name: Install modules and dependencies.
-          command: npm install --unsafe-perm
+          command: npm install
```

Comment:
Was removing `--unsafe-perm` intentional? Why is it no longer needed?

Reply:
It's intentional, we now run all tasks as a nonprivileged user (see `user: node` in each docker image description), hence `--unsafe-perm` is not needed.

Reply:
To give some more details, I changed all tasks to run as a non-privileged user a while ago because of https://github.com/ForbesLindesay/tar-pack/issues/35 which is a reason why `node-pre-gyp` is unable to unpack `grpc` binary when running as root, and `npm install` takes up to 5 minutes because it rebuilds `grpc` instead of just unpacking it. If/when `tar-pack` issue is fixed, we would be able to switch back to running as root (if we want to).

Reply:
:+1:. I have no desire to go back to running as root. I wanted to make sure that the change was intentional.